### ISSUE TITLE
[Sticky scrolling] Commented-out code is being incorrectly recognized

### ISSUE THREAD

**Issue by jakub-suliga** [eclipse-platform/eclipse.platform.ui#2128]
When code is commented out, sticky scrolling incorrectly detects the line.

To reproduce this issue:
Under org.eclipse.jdt.ui.text.folding.DefaultJavaFoldingStructureProvider.CommentPosition.findFirstContent(CharSequence, int), some code is commented out.

![image](https://github.com/user-attachments/assets/919cd9ba-3a0f-4df4-8a49-1fb7a7ad1161)

In the method org.eclipse.jdt.ui.text.folding.DefaultJavaFoldingStructureProvider.CommentPosition.computeProjectionRegions(IDocument), there is some commented-out code.
Before the the commented out line:

![image](https://github.com/user-attachments/assets/a0f32398-0371-41bf-96e8-d37d34e3c472)

And after:

![image](https://github.com/user-attachments/assets/72707c26-b4b9-45e6-9e6a-aec6c6b1930f)

**Comment by tomaswolf** [comment]
Related to https://github.com/eclipse-platform/eclipse.platform.ui/issues/2115 and https://github.com/eclipse-platform/eclipse.platform.ui/issues/1950 . Sticky scrolling would need to be able to know what a comment is, which is different in different languages. AFAIK the current implementation is based purely on indentation? 

Perhaps it could consider document partitions, if any. Though I'm not sure there is any platform-wide "standard" to identify comment partitions, and not every editor might use partitions for comments...

**Comment by Christopher-Hermann** [comment]
Correct, sticky scrolling is based purely on indentation. It is planned to provide a extension point so that every language can define a own implementation for the sticky scrolling calculation. For java we can then provide a implementation that is able to handle this kind of comments, annotations which are currently a problem or unusual code styles. 
If no specific implementation is provided, the default will still be based on indentation.

Unfortunately I had no time to look into this yet, but definitely want to pick this one up as soon as I have some time.

**Comment by Christopher-Hermann** [comment]
Fixed with #2756 and https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/2149
